### PR TITLE
refactor CI testing of dart-services

### PR DESCRIPTION
**Pull request by devoncarew**
- refactor CI testing of dart-services
- fix https://github.com/dart-lang/dart-pad/issues/2718

This PR has grown a bit but includes some important fixes.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/dart_services.yml [hunk 1]
```diff
@@ -12,56 +12,44 @@ on:
...
+  verify:
     runs-on: ubuntu-latest
     defaults:
       run:
         working-directory: pkgs/dart_services/
     strategy:
       fail-fast: false
       matrix:
-        sdk: [stable, beta, main]
+        sdk: [stable, beta, master] # main
```
Comment (devoncarew) [inline .github/workflows/dart_services.yml:27]: subosito/flutter-action doesn't seem to support main

**Comment by devoncarew** [comment]
With this PR, both `beta` and `main` are failing, but they're both failing the same way (due to https://github.com/dart-lang/dart-pad/issues/2718).

We now run `build-storage-artifacts` nightly in order to find regressions sooner.

We're working around an issue with the flutter sdk's `version` file not being created for sdks on the main branch.

In addition, it doesn't look like `subosito/flutter-action` is happy with specifying the `main` branch (we use `master` instead).

**Comment by devoncarew** [comment]
`beta` is fixed now, and `main` is past the issue with the moved flutter framework dill file.

`master` is now failing on a calls to `ui.webOnlyInitializePlatform()` - it looks like that's been removed (and perhaps is now a call to `ui_web.bootstrapEngine()`?). In any case, I think what's here is now worth review - it includes several fixes. We can work on additional fixes to the `main` channel in future PRs.

File: pkgs/dart_services/lib/src/sdk.dart [hunk 2]
```diff
@@ -96,10 +86,41 @@ class Sdk {
...
 
-String _readFile(String filePath) => File(filePath).readAsStringSync().trim();
+    // Look for 'Future<void> bootstrapEngine({ ... }) { ... }'.
+    return file.readAsStringSync().contains('bootstrapEngine(');
+  }
+
+  Map<String, dynamic> _callFlutterVersion() {
+    // Note that we try twice here as the 'flutter --version --machine' command
+    // can (erroneously) emit non-json text to stdout (for example, an initial
+    // analytics disclaimer).
```
Comment (johnpryan) [inline pkgs/dart_services/lib/src/sdk.dart:105]: Maybe we need to run `flutter config --no-analytics` and `dart --disable-analytics` somewhere?

**Review by johnpryan — APPROVED** [review]

File: pkgs/dart_services/lib/src/sdk.dart (hunk 2 above)
Comment (devoncarew) [reply pkgs/dart_services/lib/src/sdk.dart:105]: `flutter config --no-analytics` is a good call. But `flutter --version --machine` - which emits json - should never emit cruft to stdout; the output needs to be parseable.

**Comment by devoncarew** [comment]
Note that there is a test failure for `main` related to firebase. This is worth landing irrespective.

**Pull request merged**